src/frontend/src/pages/HomePage/components/TeamWorkPackageDisplay.tsx

```diff
@@ -0,0 +1,78 @@
+import WorkPackageCard from './WorkPackageCard';
+import PageBlock from '../../../layouts/PageBlock';
+import { wbsPipe } from '../../../utils/pipes';
+import Box from '@mui/material/Box';
+import { useAllTeams } from '../../../hooks/teams.hooks';
+import { useCurrentUser } from '../../../hooks/users.hooks';
+import LoadingIndicator from '../../../components/LoadingIndicator';
+import ErrorPage from '../../ErrorPage';
+import { useTheme } from '@mui/material';
+
+const TeamWorkPackageDisplay: React.FC = () => {
+  const theme = useTheme();
+  const user = useCurrentUser();
+  const { isLoading, isError, data: teams, error } = useAllTeams();
+  const myTeams = teams?.filter((team) => {
+    return (
+      team.members.some((member) => member.userId === user.userId) ||
+      team.leads.some((member) => member.userId === user.userId) ||
+      team.head.userId === user.userId
+    );
+  });
+
+  const workPackages = myTeams
+    ?.map((team) => {
+      return team.projects.map((project) => {
+        return project.workPackages;
+      });
+    })
+    .flat(2);
```

Comment:
Why do we need to flatten the return also why is myTeams?

Reply:
I used myTeams to get any work package from any project from any of the teams that a person is on. Each of these are arrays, so the type would be workPackages[][][], which is why I flatten it to just get one list of all workPackages